engine/src/execution_graph/logic_controllers/BatchJoinProcessing.h

```diff
@@ -911,21 +879,19 @@ class JoinPartitionKernel : public kernel {
 								if (i < nodes_to_send.size() - 1) {
 									worker_ids_metadata += ",";
 								}
-								node_count[nodes_to_send[i].id()]++;
+								increment_node_count(nodes_to_send[i].id(), 0);
 							}
-
 						}
-						node_count[self_node.id()]++;
-
-						ral::cache::MetadataDictionary metadata;
-						metadata.add_value(ral::cache::KERNEL_ID_METADATA_LABEL, std::to_string(this->get_id()));
-						metadata.add_value(ral::cache::QUERY_ID_METADATA_LABEL, std::to_string(this->context->getContextToken()));
-						metadata.add_value(ral::cache::ADD_TO_SPECIFIC_CACHE_METADATA_LABEL, "true");
-						metadata.add_value(ral::cache::CACHE_ID_METADATA_LABEL, small_output_cache_name);
-						metadata.add_value(ral::cache::SENDER_WORKER_ID_METADATA_LABEL, self_node.id());
-						metadata.add_value(ral::cache::WORKER_IDS_METADATA_LABEL, worker_ids_metadata);
-						ral::cache::CacheMachine* output_cache = this->query_graph->get_output_message_cache();
-						output_cache->addCacheData(std::make_unique<ral::cache::GPUCacheDataMetaData>(small_table_batch->toBlazingTableView().clone(), metadata),"",true);
+						increment_node_count(self_node.id(), 0);
+
+						send_message(std::move(small_table_batch->toBlazingTableView().clone()),
```

Comment:
here we want to use a broadcast function. We are sending the same table to every node.

Reply:
Fixed!